R/plot_fns.R

```diff
@@ -7109,6 +7143,112 @@ statres_volcano_plot <-
     return(p)
   }
 
+#' Fold change histogram for statres objects
+#'
+#' Plots a histogram indicating the number of significant
+#' biomolecules, grouped by test type and fold change direction.
+#'
+#' @param x,fc_colors passed from
+#'   \code{\link[pmartR:plot.statRes]{pmartR::plot.statRes()}}
+#' @param text_size An integer specifying the size of the text (number of
+#'   non-missing values) within the plot. The default is 3.
+#' @param display_count logical value. Indicates whether the non-missing counts will
+#'   be displayed on the bar plot. The default is TRUE.
+#' @param x_lab character string specifying the x-axis label.
+#' @param y_lab character string specifying the y-axis label.
+#' @param title_lab character string specifying the plot title.
+#' @param legend_lab character string specifying the legend title.
+#' @param show_sig logical specifying whether only significant fold changes
+#'    are included in the histogram or all fold changes
+#'
+#' @return `ggplot` object - histogram
+#'
+#' @keywords internal
+#'
+statres_histogram <- 
+  function(x,
+           fc_colors = c("blue","darkgray"),
+           text_size,
+           x_lab,
+           y_lab,
+           title_lab,
+           legend_lab,
+           free_y_axis,
+           show_sig) {
+    # If free_y_axis is true then each bar plot can have its own y axis range.
+    if (free_y_axis) {
+      da_scales <- "free"
+    } else {
+      da_scales <- NULL
+    }
+    
+    only_sig <- if (is.null(show_sig))
```

Comment:
Just checking cause the behavior changed -- we want this to default to only significant molecules?

Reply:
@leac176  if this was not the intention, the fix would be show_sig argument defaults to NULL, for histograms that turns into a TRUE and for these plots it might be FALSE